### ISSUE TITLE
OCP4 uses targetPort in some routes

### ISSUE THREAD

**Issue by sbarre-esit** [F5Networks/k8s-bigip-ctlr#1281]
#### Description
OpenShift installs with several operators that create route objects that make use of `.spec.port.targetPort` in their config. CIS doesn't seem to support this and thus the URLs are inaccessible.

```console
# oc get route --all-namespaces -o=custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name,TARGETPORT:.spec.port.targetPort | grep -v django
NAMESPACE                  NAME                     TARGETPORT
openshift-authentication   oauth-openshift          6443
openshift-console          console                  https
openshift-console          downloads                http
openshift-logging          kibana                   <none>
openshift-monitoring       alertmanager-main        web
openshift-monitoring       grafana                  https
openshift-monitoring       prometheus-k8s           web
openshift-monitoring       thanos-querier           web
```

The routes using `http` or `https` for their targetPort seem to work, but the ones using `6443` or `web` do not.

CIS log shows

```
[INFO] Port '6443' for service 'oauth-openshift' was not found.
```

#### Kubernetes Version
```console
# oc version
Client Version: openshift-clients-4.3.0-201910250623-88-g6a937dfe
Server Version: 4.3.18
Kubernetes Version: v1.16.2
```

#### Controller Version
f5networks/k8s-bigip-ctlr:1.14.0

#### BIG-IP Version
BIG-IP 14.1.2.2 Build 0.0.4 Point Release 2

#### Diagnostic Information

```console
# curl -vk https://oauth-openshift.apps.<clustername redacted>
* About to connect() to oauth-openshift.apps.<clustername redacted> port 443 (#0)
*   Trying <IP>...
* Connected to oauth-openshift.apps.<clustername redacted> (<IP>) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
* NSS error -5961 (PR_CONNECT_RESET_ERROR)
* TCP connection reset by peer
* Closing connection 0
curl: (35) TCP connection reset by peer
```

```yaml
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  labels:
    app: oauth-openshift
  name: oauth-openshift
  namespace: openshift-authentication
spec:
  host: oauth-openshift.apps.<clustername redacted>
  port:
    targetPort: 6443
  tls:
    insecureEdgeTerminationPolicy: Redirect
    termination: passthrough
  to:
    kind: Service
    name: oauth-openshift
    weight: 100
  wildcardPolicy: None
```

```yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    service.alpha.openshift.io/serving-cert-secret-name: v4-0-config-system-serving-cert
    service.alpha.openshift.io/serving-cert-signed-by: openshift-service-serving-signer@1580244275
    service.beta.openshift.io/serving-cert-signed-by: openshift-service-serving-signer@1580244275
  labels:
    app: oauth-openshift
  name: oauth-openshift
  namespace: openshift-authentication
spec:
  clusterIP: 10.x.x.x
  ports:
  - name: https
    port: 443
    protocol: TCP
    targetPort: 6443
  selector:
    app: oauth-openshift
  sessionAffinity: None
  type: ClusterIP
```

**Comment by sbarre-esit** [comment]
Here is the route/svc for Prometheus as well
```yaml
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    openshift.io/host.generated: "true"
  name: prometheus-k8s
  namespace: openshift-monitoring
spec:
  host: prometheus-k8s-openshift-monitoring.apps.<clustername redacted>
  port:
    targetPort: web
  tls:
    insecureEdgeTerminationPolicy: Redirect
    termination: reencrypt
  to:
    kind: Service
    name: prometheus-k8s
    weight: 100
  wildcardPolicy: None
```

```yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    service.alpha.openshift.io/serving-cert-secret-name: prometheus-k8s-tls
  labels:
    prometheus: k8s
  name: prometheus-k8s
  namespace: openshift-monitoring
spec:
  clusterIP: 10.x.x.x
  ports:
  - name: web
    port: 9091
    protocol: TCP
    targetPort: web
  - name: tenancy
    port: 9092
    protocol: TCP
    targetPort: tenancy
  selector:
    app: prometheus
    prometheus: k8s
  sessionAffinity: ClientIP
  sessionAffinityConfig:
    clientIP:
      timeoutSeconds: 10800
  type: ClusterIP
```

**Comment by mdditt2000** [comment]
Internal Jira for PM tracking CONTCNTR-1810

**Comment by mdditt2000** [comment]
@sbarre-esit checking on Jira this morning and the feature is currently in-progress. As of now we are looking good for a CIS 2.0

**Comment by strantalis** [comment]
This is something we need/want as well. Looking at the code this seems like it should work if you look [here](https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/pkg/appmanager/resourceConfig.go#L399). Wonder if this is more of a bug?

**Comment by mdditt2000** [comment]
@strantalis CIS didnt have the code to work with targetport. This feature will be added to CIS 2.1.

**Comment by mdditt2000** [comment]
Wrapping up CIS 2.1. Closing this issue out as issue is done. CIS 2.1 will be available middle of July. TY

**Comment by phbergsmann** [comment]
Hi!

While trying to dig deeper into my issues #1806 I found the same line in the log output as shown in thisbug report

[INFO] [CORE] Port '6443' for service 'oauth-openshift' was not found.

So I guess my console login might not work since the targetPort: 6443 is not resolved. I'm currently using the latest version of the controller (2.4.0). I've also tried to downgrade to 2.1.1 but the same log line appeared
